File: sw-python/logger.py
import coloredlogs, logging

class Logger:
    def __init__(self, module_name):
        coloredlogs.install(fmt='[%(asctime)-15s] %(name)s %(message)s')
        logging.basicConfig(level=logging.NOTSET)
        self.logger = logging.getLogger(module_name)

    def info(self, message):
        self.logger.info(message)

    def warning(self, message):
        self.logger.warning(message)

    def error(self, message):
        self.logger.error(message)

File: sw-cpp/logger.cpp
#include "logger.hpp"

Logger::Logger(std::string module_name) {
    name = module_name;
}

void Logger::info(std::string message) {
    std::cout << "[" + name + "]  INFO: " << message << std::endl;
}

void Logger::warning(std::string message) {
    std::cout << "[" + name + "]  WARN: " << message << std::endl;
}

void Logger::error(std::string message) {
    std::cout << "[" + name + "] ERROR: " << message << std::endl;
}
